### PR TITLE
Change array used to check for dup ids into a Set

### LINKED ISSUE #21: Change the array used to check for duplicate ids into a set (edited issue title to reflect the change we decided on)
When checking for duplicated IDs we push existing ids into an array: https://github.com/indeedeng/starfish/blob/dc651c846bbbe1bc6eefee70a24cdf44383c59a2/index.js#L35

Changing the data structure to an object will allow for O(1) lookup time instead of O(n) having to look through the whole array to check which will speed up the process and scale a little better. It could also reduce/simply the logic needed to identify duplicated github ids.

### PR DESCRIPTION
**Pull request by soulchainer**
Closes #21.

And refactor of the whole file.
Tests are passing and all that :).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,139 +1,158 @@
...
+  arrayOfIdObjects.push(idObject);
 }
 
 function filterContributorByTime(idObject, dates) {
-  const timeWindow = moment.range([moment.utc(`${dates[0]}`, 'YYYY-MM-DD hh:mm'), moment.utc(`${dates[1]}`, 'YYYY-MM-DD hh:mm')]);
+  const timeWindow = moment.range([
+    moment.utc(`${dates[0]}`, "YYYY-MM-DD hh:mm"),
+    moment.utc(`${dates[1]}`, "YYYY-MM-DD hh:mm")
+  ]);
   const contribsByUser = [];
```
Comment (soulchainer) [inline index.js]: I didn't remove this `const`, just in case, but: what it's really doing? Is not being used in any place. Shouldn't it also be removed?

**Comment by soulchainer** [comment]
Update: solved conflicts that appeared because of another PR was merged.

**Comment by danisyellis** [comment]
Hi @soulchainer . I'm so sorry I didn't respond to this PR when you put it in. I appreciate the time and effort you took to work on this issue for Starfish.
I was very new to being a maintainer and was overwhelmed by having so many changes (practically every line in the file) in one commit, some that I wanted and others that I didn't, and I wasn't sure how to handle it.
Since then, we've added a linter to fix a lot of that. And I've gotten more experience handling PRs.
Would you like to take a look at the current code and make a new Pull Request to handle issue #21 (Change the array used to check for duplicate ids into a set)?

**Comment by soulchainer** [comment]
No worries, @danisyellis :).
I will check it after I end my working day, I hope :).
Enjoy your friday ;).

**Comment by danisyellis** [comment]
@soulchainer I was thinking, since the actual change you're making (changing an array into a set) is just a few lines of code, it might be easiest to close this PR and create a new one to close out #21 
This PR is large and involved, and there have been a lot of changes to the codebase between then and now. 

So, unless you object, I'm going to close out this PR. Would you like to make the new one?

**Comment by danisyellis** [comment]
Since we haven't heard from you, someone on my team is going to make this change. 

Sorry again for the confusion. I learned things, if that helps :-)

**Pull request closed without merging**